File: ts/project/.sdk/tm/js/src/Response.js

const { getprop } = require('./utility/StructUtility')


class Response {
  constructor(resmap) {
    this.status = getprop(resmap, 'status', -1)
    this.statusText = getprop(resmap, 'statusText', '')
    this.headers = getprop(resmap, 'headers')
    this.json = resmap.json ? resmap.json.bind(resmap) : async () => undefined
    this.body = getprop(resmap, 'body')
    this.err = getprop(resmap, 'err')
  }
}


module.exports = {
  Response,
}

File: ts/project/.sdk/tm/swift/Sources/ProjectNameSDK/core/Response.swift
// ProjectName SDK - transport response wrapper.

import Foundation

public final class Response {
  public var status: Int = -1
  public var statusText: String = ""
  public var headers: Value = .noval
  public var jsonFunc: NativeCall0? = nil
  public var body: Value = .noval
  public var err: Error? = nil

  public init(_ resmap: VMap?) {
    let m: Value = resmap == nil ? .map(VMap()) : .map(resmap!)

    let st = getprop(m, .string("status"))
    if !isNil(st) { status = toInt(st) }

    if let s = getprop(m, .string("statusText")).asString { statusText = s }

    headers = getprop(m, .string("headers"))

    if let jf = getprop(m, .string("json")).asNative as? NativeCall0 { jsonFunc = jf }

    body = getprop(m, .string("body"))

    if let e = getprop(m, .string("err")).asNative as? Error { err = e }
  }
}
